examples/PlaneCurve.jl

```diff
@@ -0,0 +1,273 @@
+  end
+  return(C.degree)
+end
+
+################################################################################
+
+@doc Markdown.doc"""
+    jacobi_ideal(C::PlaneCurve)
+
+Return the Jacobian ideal of the defining polynomial of `C`.
+"""
+function Oscar.jacobi_ideal(C::PlaneCurve)
+ return jacobi_ideal(C.eq)
+end
+
+################################################################################
+# helping function: computes the irreducible components of the curve
+
+function compo(C::PlaneCurve)
+  if typeof(C) == AffinePlaneCurve{typeof(lc(C.eq))}
+    D = factor(C.eq)
+    C.components = Dict(AffinePlaneCurve(x) => D.fac[x] for x in keys(D.fac))
+  elseif typeof(C) == ProjPlaneCurve{typeof(lc(C.eq))}
+    D = factor(C.eq.f)
+    C.components = Dict(ProjPlaneCurve(x) => D.fac[x] for x in keys(D.fac))
+  end
+end
+
+################################################################################
+# Components of the curve
+
+@doc Markdown.doc"""
+    curve_components(C::PlaneCurve)
+
+Return a dictionary containing the irreducible components of `C` and their multiplicity.
+"""
+function curve_components(C::PlaneCurve)
+  if isempty(C.components)
+    compo(C)
+  end
+  return C.components
+end
+
+################################################################################
+# Check irreducibility.
+# TODO: change for a direct irreducibility check when available.
+
+@doc Markdown.doc"""
+    isirreducible(C::PlaneCurve)
+
+Return `true` if `C` is irreducible, and `false` otherwise.
+"""
+function Oscar.isirreducible(C::PlaneCurve)
+  if isempty(C.components)
+     compo(C)
+  end
+  if length(C.components) != 1
+     return false
+  else
+     return all(isone, values(C.components))
+  end
+end
+
+################################################################################
+# Check reducedness by computing a factorization
+# TODO: change for a direct squarefree check when available.
+
+@doc Markdown.doc"""
+    isreduced(C::PlaneCurve)
+
+Return `true` if `C` is reduced, and `false` otherwise.
+"""
+function Oscar.isreduced(C::PlaneCurve)
+  if isempty(C.components)
+    compo(C)
+  end
+  return all(isone, values(C.components))
+end
+
+################################################################################
+# Compute squarefree equation.
+# TODO: change for a direct squarefree computation when available.
+
+@doc Markdown.doc"""
+    reduction(C::PlaneCurve)
+
+Return the plane curve defined by the squarefree part of the equation of `C`.
+"""
+function reduction(C::PlaneCurve)
+  if isempty(C.components)
+     compo(C)
+  end
+  F = prod(D -> D.eq, keys(C.components))
+  if typeof(C) == AffinePlaneCurve{typeof(lc(C.eq))}
+    rC = AffinePlaneCurve(F)
+    rC.components = Dict(AffinePlaneCurve(D.eq) => 1 for D in keys(C.components))
+  elseif typeof(C) == ProjPlaneCurve{typeof(lc(C.eq))}
+    rC = ProjPlaneCurve(F)
+    rC.components = Dict(ProjPlaneCurve(D.eq) => 1 for D in keys(C.components))

```

Comment:
```suggestion
    rC.components = Dict(ProjPlaneCurve(D.eq.f) => 1 for D in keys(C.components))

```

Reply:
Thanks for the comments. By forbidding the defining equation to be constant, I forbid also curves without components. Regarding your suggestion to write C.components = Dict(T(x) => e for (x, e) in D)),  it does not work, T(x) does not create a plane curve, it throws the error "ERROR: MethodError: no method matching ProjPlaneCurve{fmpq}(::fmpq_mpoly)". Is there a way to fix it ?

Reply:
Ah, I see. Yes, there is a way, but is a bit more intrusive, so I would ignore it for this PR here.